modules/fswatch/fswatch.go

```diff
@@ -0,0 +1,215 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Contributor: Aaron Meihm ameihm@mozilla.com [:alm]
+
+package fswatch /* import "mig.ninja/mig/modules/fswatch" */
+
+import (
+	"encoding/json"
+	"fmt"
+	"runtime"
+
+	"mig.ninja/mig/modules"
+)
+
+type module struct {
+}
+
+func (m *module) NewRun() modules.Runner {
+	return new(run)
+}
+
+func init() {
+	modules.Register("fswatch", new(module))
+}
+
+type run struct {
+	Parameters Parameters
+	Results    modules.Result
+}
+
+func buildResults(e elements, r *modules.Result) (buf []byte, err error) {
+	r.Success = true
+	r.Elements = e
+	r.FoundAnything = true
+	buf, err = json.Marshal(r)
+	return
+}
+
+var logChan chan string
+var handlerErrChan chan error
+var configChan chan []byte
+
+var alertChan chan FSWatchAlert
+
+// Alert severities
+const (
+	_ = iota
+	ALERT_CRITICAL
+	ALERT_HIGH
+	ALERT_MEDIUM
+	ALERT_LOW
+)
+
+// Create a new alert and send it to channel ch
+func newAlert(sev int, f string, a ...interface{}) {
+	alertChan <- FSWatchAlert{
+		Severity: sev,
+		Alert:    fmt.Sprintf(f, a...),
+	}
+}
+
+// An alert generated by the fswatch module
+type FSWatchAlert struct {
+	Severity int    `json:"severity"` // ALERT_CRITICAL, etc...
+	Alert    string `json:"alert"`    // alert text
```

Comment:
nit on naming: the module is already called `fswatch` so no need to repeat it in the struct name. Also, the `Alert` field could be more descriptive, like `Message` or `Body`.
```go
type Alert struct {
    Severity int
    Message string
}
```